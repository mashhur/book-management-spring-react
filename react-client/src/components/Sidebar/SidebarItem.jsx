import { BiChevronRight } from "react-icons/bi";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

function SidebarItem({ name, url }) {
	return (
		<div
			data-test="component-side-bar-item"
			style={{
				borderRadius: "2.3rem",
				fontSize: "1.0rem",
				marginBottom: ".5rem",
			}}
			className="col">
			<Link
				to={url ? url : `/cat/${name}`}
				className="list-group-item d-flex 
				justify-content-between 
				align-items-center 
				list-group-item-action 
				py-3 px-4 bg-category fs-6 ">
				{name}
				<span>
					<BiChevronRight
						style={{ fontSize: "23px", opacity: 0.6 }}
					/>
				</span>
			</Link>
		</div>
	);
}

SidebarItem.propType = {
	name: PropTypes.string.isRequired,
	url: PropTypes.string,
};

export default SidebarItem;
