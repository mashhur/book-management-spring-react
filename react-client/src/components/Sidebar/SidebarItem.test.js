import React from "react";
import { shallow } from "enzyme";
import SidebarItem from "./SidebarItem";
import {
	findByTestAttr,
	checkProp,
} from "../../tests/testUtils";
import { Link } from "react-router-dom";

const defaultProps = { name: "Horror" };

const setup = (props = {}) => {
	const setUpProps = { ...defaultProps, ...props };
	return shallow(<SidebarItem {...setUpProps} />);
};

describe("Testing side nav component", () => {
	it("renders without any error", async () => {
		const wrapper = setup();
		const component = await findByTestAttr(
			wrapper,
			"component-side-bar-item"
		);
		expect(component.length).toBe(1);
	});
	describe("should have the correct navigation when", () => {
		it("`url` not passed", async () => {
			const wrapper = setup();
			const component = await findByTestAttr(
				wrapper,
				"component-side-bar-item"
			);
			expect(component.find(Link).prop("to")).toBe(
				`/cat/${defaultProps.name}`
			);
		});
		it("`url`was passed", async () => {
			const wrapper = setup({
				url: "/",
				name: "All Books",
			});
			const component = await findByTestAttr(
				wrapper,
				"component-side-bar-item"
			);
			expect(component.find(Link).prop("to")).toBe("/");
		});
	});

	it("doesnot throw warning with expected props", async () => {
		checkProp(SidebarItem, defaultProps);
	});
});
