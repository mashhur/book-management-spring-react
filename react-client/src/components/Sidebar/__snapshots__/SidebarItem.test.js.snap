// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing side nav component should match the default snapshot 1`] = `ShallowWrapper {}`;
